### PR TITLE
feat(flutter): add custom attribute documentation

### PR DESCRIPTION
**Pull request by haverchuck**
#### Description of changes:
Adds documentation for custom attributes in Amplify Flutter.

#### Related GitHub issue #, if available:
https://github.com/aws-amplify/amplify-flutter/issues/1180

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [X] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [X] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [X] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Custom attributes
+Amplify Flutter supports [standard OIDC user attributes](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims) as well as custom attributes. Custom attributes can be instantiated via the custom attribute constructor:
```
Comment (abdallahshaban557) [inline src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx:2]: do we want to mention OIDC here - or mention the standard attributes from Cognito?

Comment (haverchuck) [reply src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx:2]: They are the same, I believe.

Comment (abdallahshaban557) [reply src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx:2]: I think it would make sense to link to Cognito's attributes

Comment (haverchuck) [reply src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx:2]: Oh I see - yeah, sure.

File: src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    username: username,
+    password: password,
+    options: SignUpOptions(
+      userAttributes: userAttributes,
+    ),
+  );
+}
+```
+
+When working with a Cognito UserPool, you can setup [custom attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes) via the Cognito console or AWS CLI. Although Cognito prepends a "custom:" prefix on the attribute name, there is no need for you to add this in Amplify Flutter's custom attribute constructor.
```
Comment (dnys1) [inline src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx:26]:
```suggestion
When working with a Cognito UserPool, you can set up [custom attributes](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#user-pool-settings-custom-attributes) via the Cognito console or AWS CLI. Although Cognito prepends a "custom:" prefix on the attribute name, there is no need for you to add this in Amplify Flutter's custom attribute constructor.
```

File: src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx (hunk 1 above)
Comment (dnys1) [reply src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx:2]: I'm okay either way. Cognito links to OIDC as well: https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-attributes.html#cognito-user-pools-standard-attributes

**Review by dnys1 — APPROVED** [review]

File: src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx (hunk 1 above)
Comment (abdallahshaban557) [reply src/fragments/lib/auth/flutter/user_attributes/50_custom_attributes.mdx:2]: hmmm - good point. Let us keep it as is for now.

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by Jordan-Nelson — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**